owncloudApp/src/main/java/com/owncloud/android/usecases/synchronization/SynchronizeFolderUseCase.kt

```diff
@@ -51,6 +53,7 @@ class SynchronizeFolderUseCase(
                             accountName = accountName,
                             spaceId = ocFile.spaceId,
                             syncMode = params.syncMode,
+                            isActionSetFolderAvailableOffline = params.isActionSetFolderAvailableOffline,
```

Comment:
It needs to be on both sides. When we encounter a subfolder scenario, if we do not continue setting the `isActionSetFolderAvailableOffline` parameter it will default to `false`, when in reality we are setting all content available offline. Then the content of these subfolders will not be downloaded or synchronized.

Reply:
Yes, but this should only apply to files, right? Passing it as parameter to folders is just needed to transmit it until we arrive to a file, and we're doing it in the recursive call. In line 44 we're passing it to a refresh, where we don't need to check the new condition. Am I missing something?

Reply:
In the case you are describing, if we leave the parameter isActionSetFolderAvailableOffline to default (false), if we have a subfolder inside the parent folder, which does not meet any condition of this if:
```
else if (localChildToSync.etag != remoteChild.etag ||
localChildToSync.localModificationTimestamp > remoteChild.lastSyncDateForData!! ||
isActionSetFolderAvailableOffline
)
```
It will not save the folder in the folderContentUpdated and therefore when it does the recursion the files that are in that folder will no longer be taken into account.